lib/charms/nginx_ingress_integrator/v0/ingress.py

```diff
@@ -117,7 +125,22 @@ class IngressRequires(Object):
     def __init__(self, charm, config_dict):
         super().__init__(charm, "ingress")
 
-        self.framework.observe(charm.on["ingress"].relation_changed, self._on_relation_changed)
+        self.framework.observe(
+            charm.on["ingress"].relation_changed, self._on_relation_changed
+        )
+
+        # Set default values.
+        DEFAULT_RELATION_FIELDS = {
+            "service-namespace": self.model.name,
+        }
+        for k, v in DEFAULT_RELATION_FIELDS.items():
+            if k not in config_dict or not config_dict[k]:
+                config_dict[k] = v
+
+        # And now populate data for conformity with charm-relation-interfaces.
+        for k, v in RELATION_INTERFACES_MAPPINGS.items():
+            if k in config_dict and config_dict[k]:
```

Comment:
Do we need to check the existence of the key given this is validated afterwards in the `_config_dict_errors` method?

Reply:
I don't quite follow, this step is performing a mapping of the old keys to the new keys defined in the ingress standard to ensure compatibility with both. The code isn't very clear, I'll try and update it to make it clearer.